pkgs/by-name/ou/outfly/package.nix

```diff
@@ -0,0 +1,96 @@
+{
+  lib,
+  fetchFromGitea,
+  buildType ? "release",
+  platform ? "linux",
+  enableWayland ? true,
+  enableXorg ? true,
+  enableOpenGL ? true,
+  enableVulkan ? true,
+  rustPlatform,
+  makeDesktopItem,
+  pkg-config,
+  libxkbcommon,
+  alsa-lib,
+  libGL,
+  vulkan-loader,
+  wayland,
+  libXrandr,
+  libXcursor,
+  libX11,
+  libXi,
+  ...
+}:
+
+assert (platform == "linux") -> (enableWayland || enableXorg);
+assert (platform == "linux") -> (enableVulkan || enableOpenGL);
+
+rustPlatform.buildRustPackage rec {
+  pname = "outfly";
+  inherit buildType;
+  version = "0.10.1";
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "outfly";
+    repo = "outfly";
+    rev = "v${version}";
+    hash = "sha256-VmUiDORTlXQvAKNXSQOI4+xqp/uhqBcKGHdzhM+D1pU=";
+
+  };
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [
+    "embed_assets"
+  ] ++ lib.optionals enableXorg [ "x11" ] ++ lib.optionals enableWayland [ "wayland" ];
+
+  runtimeInputs =
+    [ libxkbcommon ]
+    ++ lib.optionals enableOpenGL [ libGL ]
+    ++ lib.optionals enableXorg [
+      libXrandr
+      libX11
+    ]
+    ++ lib.optionals enableVulkan [ vulkan-loader ];
+
+  buildInputs = [
+    alsa-lib.dev
+    libXcursor
+    libXi
+  ] ++ lib.optionals enableWayland [ wayland ];
+
+  nativeBuildInputs = [ pkg-config ];
+  doCheck = false;
+
+  postFixup = ''
+    patchelf $out/bin/outfly \
+    --add-rpath ${lib.makeLibraryPath runtimeInputs}
+  '';
+
+  postPatch = ''
+    find ../cargo-vendor-dir -name Cargo.toml -exec sed -i 's/^workspace = true//' {} +
+  '';
```

Comment:
is far as i can tell this is unnecessary. (and fails when using `cargoHash`)

Reply:
thank you, that is an artifact from an earlier version thatsome repos from outside crates.io. 
and nixpkgs rust vendor scripts behaved incompatibly to cargo-vendor.